testeng/jobs/backupJenkins.groovy

```diff
@@ -75,13 +86,14 @@ secretMap.each { jobConfigs ->
             env('AWS_DEFAULT_REGION', jobConfig['region'])
         }
         
+        // Sync currently paged files to disk
+        String script = "sync\n"
         // This might seem overkill, but in case the pip requirements change, read them from
         // the requirements file in the workspace
-        String script = ""
         readFileFromWorkspace('testeng/resources/requirements.txt').split("\n").each { line ->
-            script += "pip install ${line}; "
+            script += "pip install --exists-action w ${line}\n"
         }
-        script += "aws ec2 create-snapshot --volume-id ${jobConfig['volumeId']} --description 'Automatic ${jobConfig['jenkinsInstance']} jenkins snapshot'"
+        script += "aws ec2 create-snapshot --volume-id ${jobConfig['volumeId']} --description 'Automatic ${jobConfig['jenkinsInstance']} jenkins snapshot' |tee \${WORKSPACE}/snapshot-out"
```

Comment:
Piping to tee will lose the exit code if the left half of the command fails.
Maybe take a look at [this stack overflow article](http://stackoverflow.com/questions/1221833/bash-pipe-output-and-capture-exit-status) for some options.